webrtc.html

```diff
@@ -261,6 +262,12 @@ <h2>Dictionary <a class="idlType">RTCConfiguration</a> Members</h2>
                 <span data-jsep=
                 "ice-candidate-pool constructor">[[!JSEP]]</span>.</p>
               </dd>
+              <dt><dfn data-idl><code>adaptivePtime</code></dfn> of type <span class="idlMemberType">boolean</span>, defaulting to <code>false</code>.</dt>
+              <dd>
+                <p>Indicates whether audio senders may dynamically change the frame lengths so as to adapt to changing network conditions.
```

Comment:
I'd like to clarify what the UA is or is not allowed to do, it sounds like if the value is true then changing the length is optional and up to the UA (is it a MAY or a SHOULD, would it be allowed to change the value for reasons other than "network conditions", how or when would it be allowed to change it, etc?), whereas if the value is false it sounds like it MUST NOT change. So I'd say that it MAY change on a per-packet basis, i.e. it's entirely up to the implementation, and that we don't mandate what the cause of the change is.

It should also be clear that it MUST use the same length if the value was false, otherwise there's no difference between on or off. Still, I don't think different codecs would necessarily use the same frame length, so I would tie deciding the "constant" frame length to be done when negotiating and say that it is the same per sender-and-codec. This is when we ensure resources are available as well, so it would makes sense there I guess.

I still want to mention network conditions since that is the motivation for the API, but considering it is entirely up to the UA on how it decides to change the length, I'd put this in a note rather than in a description about how the UA implements it.

Suggestion:

> Indicates whether audio RTCRtpSenders may dynamically change the frame lengths. If this value is true, the User Agent may use any valid frame length (less than the negotiated maxptime) for any of its packets, and MAY use a different frame length on a per-packet basis. If this value is false, the User Agent MUST decide the frame length to use for a particular RTCRtpSender and codec at time of negotiation, and MUST NOT change the frame length used for that RTCRtpSender-codec pair unless another neogitation is performed.
> [Note: Higher frame lengths reduce the bandwidth consumption due to overhead, but do so at the cost of increased latency. Allowing the User Agent to change the frame length dynamically allows it to adapt it's strategy, for example, based on network conditions.]

Reply:
I've taken part of your suggestion. PTAL.